osidb/models.py

```diff
@@ -1808,6 +1845,76 @@ class Meta:
             "created_dt",
         )
 
+    def bzsync(self, *args, bz_api_key=None, **kwargs):
+        """
+        Bugzilla sync of the FlawComment instance and of the related Flaw instance.
+        """
+
+        self.save()
+
+        # Comments need to be synced through flaw
+        # If external_system_id is blank, BugzillaSaver posts the new comment
+        # and FlawCollector loads the new comment and updates this FlawComment
+        # instance to match bugzilla.
+        self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
+
+    def _validate_single_pending_comment(self):
+        """
+        Check that this is at most the first pending comment for the flaw.
+        This comment must not be a second pending comment for the flaw. If it
+        is, it means that API calls are not ATOMIC_REQUESTS anymore, or that
+        the temporary pending comments are not cleaned up properly, or that
+        something makes concurrent requests.
+        """
+        tmp_cmnts = FlawComment.objects.pending().filter(flaw=self.flaw)
+        if tmp_cmnts.count() > 1:
+            raise ValidationError(
+                f"FlawComment ({self.uuid}) for {self.flaw} is not the first "
+                "pending new comment."
+            )
+
+    def _validate_comment_order(self):
+        """
+        Check that this comment's order (comment number) is directly following
+        the last existing comment of the given flaw. This checks that the
+        client has a sufficiently up-to-date idea of the flaw when submitting
+        the comment.
+
+        Usually, a flaw has a comment number 0 as its description. However,
+        if the description is missing, the first non-description comment must
+        start with number 1.
+        """
+        if self.flaw.comments.count() == 0:
+            proper_following_order = 1
+        else:
+            proper_following_order = (
+                1 + self.flaw.comments.aggregate(Max("order"))["order__max"]
+            )
+        if self.external_system_id == "" and proper_following_order != self.order:
+            raise ValidationError(
+                f"FlawComment ({self.uuid}) for {self.flaw} is not of the correct "
+                f"order. FlawComment.order ({self.order}) but flaw has "
+                f"({self.flaw.comments.count()}) comments (starting with 0). "
+                "Either OSIDB's DB is outdated or the client has an outdated "
+                "idea of the Flaw."
+            )
+
+    def _validate_nonblank_extid_for_nonpending(self):
```

Comment:
This should not be validated here. The user can do nothing about it and would probably not even understand what is wrong. If we have a blank external ID of an already collected (from Bugzilla) comment it is not even a data error but a bug. I am also not sure why do we operate here with 1 minute. The flaw (and all the related data) is force synced after every update so we do not have to wait for the periodic sync to have it up-to-date.

Reply:
The checks are here basically to prevent invalid objects to be saved during concurrent operations, but it turns out it doesn't work.

I'll delete it.

Related to discussion https://github.com/RedHatProductSecurity/osidb/pull/209/#discussion_r1209214388 (`https://github.com/RedHatProductSecurity/osidb/pull/209/#discussion_r1209214388`)